lib/testcontainers.ex

```diff
@@ -231,13 +231,31 @@ defmodule Testcontainers do
       |> Container.with_label(container_lang_label(), container_lang_value())
       |> Container.with_label(container_label(), "#{true}")
 
-    with {:ok, _} <- Api.pull_image(config.image, state.conn, auth: config.auth),
-         {:ok, id} <- Api.create_container(config, state.conn),
-         :ok <- Api.start_container(id, state.conn),
-         {:ok, container} <- Api.get_container(id, state.conn),
-         :ok <- ContainerBuilder.after_start(config_builder, container, state.conn),
-         :ok <- wait_for_container(container, config.wait_strategies || [], state.conn) do
-      {:ok, container}
+    hash = :crypto.hash(:sha256, :erlang.term_to_binary(Map.from_struct(config))) |> Base.encode16()
```

Comment:
it seems to work pretty well, but i have a hunch to make some unit tests for this, after moving this to a util module